mlog/log.go

```diff
@@ -73,6 +74,7 @@ type Logger struct {
 	consoleLevel zap.AtomicLevel
 	fileLevel    zap.AtomicLevel
 	logrLogger   *logr.Logger
+	mutex        *sync.RWMutex
```

Comment:
There is no need to use a pointer here. The receiver is already a pointer, so that should be sufficient.

Reply:
I used a pointer here because we do a deep copy on the Shutdown which will leak resources right?

Reply:
Ah I see. Indeed I missed that. Thanks!